java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -2910,6 +3047,18 @@ public final ColumnVector replaceRegex(String pattern, Scalar repl) {
     return replaceRegex(pattern, repl, -1);
   }
 
+  /**
+   * For each string, replaces any character sequence matching the given regex program pattern
+   * using the replacement string scalar.
+   *
+   * @param regexProg The regex program with pattern to search within each string.
+   * @param repl The string scalar to replace for each pattern match.
+   * @return A new column vector containing the string results.
+   */
+  public final ColumnVector replaceReRegexProg(RegexProgram regexProg, Scalar repl) {
```

Comment:
Instead of adding in a `RegexpProg` to the end of the methods, I think it would be better to remove the `Re` at the end and rely on the `RegexProgram` type to allow for method overloading.

```
public final ColumnVector replace(RegexProgram regexProg, Scalar repl) {
```

We then should deprecate the old APIs and implement them in terms of the new API. You need to be careful that we get the capture groups correct for the default because we want the default performance to be good. I don't think we have to worry about capture groups except for replace with backrefs and the extract APIs.

This goes for all of the APIs being added.  Internally the native APIs cannot have overloading so it is fine to have the longer name.